dev/io.openliberty.org.jboss.weld4/test/io/openliberty/weld/jsf/WeldJSFCheckSumTest.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.weld.jsf;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+/**
+ *  Tests to look fo changes in FacesUrlTransformer & ConversationAwareViewHandler.
+ *  If the checksum changes, then the latest classes should be ported over to com.ibm.ws.jsfContainer.
+ *  See issue https://github.com/OpenLiberty/open-liberty/issues/14524
+ */
+public class WeldJSFCheckSumTest {
+
+    File bundle;
+
+    String bundleFileName = "io.openliberty.org.jboss.weld4.jar";
+
+    String pathToFacesUrlTransformer = "org/jboss/weld/module/jsf/FacesUrlTransformer.class";
+    String pathToConversationAwareViewHandler= "org/jboss/weld/module/jsf/ConversationAwareViewHandler.class";
+
+    @Test
+    public void testCheckSumForFacesUrlTransformer() {
+      long expectedCheckSum  =  1823966370L; // Current as of Nov 6th, 2020
+      long actualCheckSum = getChecksum(getPathToWeldBundle(), pathToFacesUrlTransformer);
+      assertEquals("The checksum forFacesUrlTransformer has changed! " +
+                    "Please notify owners of com.ibm.ws.jsfContainer!", expectedCheckSum, actualCheckSum);
```

Comment:
Perhaps make this error message a bit more specific like:

> "The checksum for " + pathToFacesUrlTransformer  + " has changed! Whatever change was made to this file needs to be investigated and replicated in <path to jsfContainer copy of the class> and then the checksum for this test method can be updated to " + actualChecksum

Reply:
let me upate